src/tests/ngraph_helpers/lpt_ngraph_functions/src/gather_function.cpp

```diff
@@ -0,0 +1,102 @@
+
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/opsets/opset8.hpp>
+#include "lpt_ngraph_functions/common/builders.hpp"
+
+namespace ngraph {
+namespace builder {
+namespace subgraph {
+
+std::shared_ptr<ngraph::Function> GatherFunction::getOriginal(
+    const ngraph::PartialShape& inputShape,
+    const std::vector<size_t>& gatherIndicesShape,
+    const std::vector<int>& gatherIndicesValues,
+    const std::vector<int>& axis,
+    const int64_t batch_dims,
+    const ngraph::element::Type precisionBeforeDequantization,
+    const ngraph::builder::subgraph::DequantizationOperations& dequantization,
+    const int opset_version) {
+    const auto input = std::make_shared<ngraph::opset1::Parameter>(precisionBeforeDequantization, inputShape);
+    const std::shared_ptr<Node> dequantizationOp = makeDequantization(input, dequantization);
+    const auto indicesNode = std::make_shared<ngraph::opset1::Constant>(
+        ngraph::element::i64,
+        ngraph::Shape(gatherIndicesShape),
+        gatherIndicesValues);
+    const auto axisNode = std::make_shared<ngraph::op::Constant>(ngraph::element::i64, ngraph::Shape{ axis.size() }, axis);
+    const std::shared_ptr<Node> gather = std::make_shared<ngraph::opset8::Gather>(
+        dequantizationOp, indicesNode, axisNode, batch_dims);
+    gather->set_friendly_name("output");
+
+    ngraph::ResultVector results{ std::make_shared<ngraph::opset1::Result>(gather) };
+    return std::make_shared<ngraph::Function>(results, ngraph::ParameterVector{ input }, "GatherFunction");
+}
+
+std::shared_ptr<ngraph::Function> GatherFunction::getOriginal(
+    const ngraph::PartialShape& inputShape,
+    const std::vector<size_t>& gatherIndicesShape,
+    const std::vector<int>& gatherIndicesValues,
+    const std::vector<int>& axis,
+    const int64_t batch_dims,
+    const ngraph::element::Type precisionBeforeFq,
+    const FakeQuantizeOnData& fqOnData,
+    const int opset_version) {
+    const auto input = std::make_shared<ngraph::opset1::Parameter>(precisionBeforeFq, inputShape);
+
+    const std::shared_ptr<Node> quantizationOp = fqOnData.empty() ?
+        std::dynamic_pointer_cast<ngraph::Node>(input) :
+        makeFakeQuantize(input, precisionBeforeFq, fqOnData);
+
+    const auto indicesNode = std::make_shared<ngraph::opset1::Constant>(
+        ngraph::element::i64,
+        ngraph::Shape(gatherIndicesShape),
+        gatherIndicesValues);
+    const auto axisNode = std::make_shared<ngraph::opset1::Constant>(ngraph::element::i64, ngraph::Shape{ axis.size() }, axis);
+    const std::shared_ptr<Node> gather = std::make_shared<ngraph::opset8::Gather>(quantizationOp, indicesNode, axisNode, batch_dims);
+
+    ngraph::ResultVector results{ std::make_shared<ngraph::opset1::Result>(gather) };
+    return std::make_shared<ngraph::Function>(results, ngraph::ParameterVector{ input }, "GatherFunction");
+}
+
+std::shared_ptr<ngraph::Function> GatherFunction::getReference(
```

Comment:
Why you don't use `makeFakeQuantize` here? Did you lost decomposed `FakeQuantize` in reference function?

Reply:
This getReference function is the reference of the first getOriginal function which has the subgraph input-->dequantization-->gather-->result, after the transformation, the subgraph should be the reference function:  -->gather-->dequantization-->result, so we don't need makeFakeQuantize in the reference function.